Airline order course day1 start

joekyou opened the pull request:
测试通过，参照：TestReadMe.md
功能简介，参照：README.md

服务启动后（cd airline-order-course/backend && mvn spring-boot:run）
http://localhost:8080/swagger-ui/index.html

备注：18.207.201.92:3306  要启动，我测试通过了，但是我提交前，这个DB已经被关闭了 ：（

Angular前端工程搭建。-》 没有实施
CI/CD与云端部署          -》 没有创建
前端Docker化                -》 没有实施
后端Docker化                -》 没有实施
Docker Compose 编排服务 -》 没有实施


完成的主要功能：
已完成的功能模块
1. 数据层 (Entity & Repository)
* 用户实体 (User.java) - 用户信息管理
* 订单实体 (Order.java) - 订单信息管理，包含完整的状态机
* 航班信息实体 (FlightInfo.java) - 航班信息管理
* 枚举类 - OrderStatus、FlightStatus、UserRole
* Repository层 - 完整的数据访问层，包含复杂查询
2. 业务层 (Service)
* 用户服务 (UserService.java) - 用户管理业务逻辑
* 认证服务 (AuthService.java) - JWT认证、登录登出
* 订单服务 (OrderService.java) - 订单管理、状态机处理
* 航班服务 (FlightInfoService.java) - 航班信息管理
3. 控制层 (Controller)
* 认证控制器 (AuthController.java) - 登录、注册、令牌管理
* 订单控制器 (OrderController.java) - 订单CRUD、状态更新
* 航班控制器 (FlightInfoController.java) - 航班信息管理
4. 安全配置
* JWT工具类 (JwtUtil.java) - JWT令牌生成和验证
* 安全配置 (SecurityConfig.java) - Spring Security配置
* JWT过滤器 (JwtRequestFilter.java) - 请求拦截和认证
5. 数据传输对象 (DTO)
* 用户DTO - 用户信息传输
* 订单DTO - 订单信息传输
* 航班DTO - 航班信息传输
* 认证请求/响应DTO - 登录认证数据传输
6. 配置文件
* 数据库配置 - 连接到你的MySQL数据库
* JPA配置 - Hibernate配置优化
* Swagger配置 - API文档自动生成
核心功能特性
用户认证系统
* JWT令牌认证
* 用户注册/登录
* 角色权限控制 (USER/ADMIN)
* 令牌刷新和验证
订单管理系统
* 完整的订单状态机：
    * PENDING_PAYMENT → PAID → TICKETING_IN_PROGRESS → TICKETED
    * 支持取消和失败状态
* 订单创建、查询、更新、取消
* 座位预订和释放机制
* 超时订单自动取消
航班管理系统
* 航班信息CRUD操作
* 多条件搜索 (航线、时间、价格)
* 座位管理和可预订状态
* 航班状态管理
权限控制
* 用户只能操作自己的订单
* 管理员可以管理所有数据
* 公开API (航班查询) 无需认证
数据库设计
已有表结构
* app_users_qiaozhe - 用户表
* orders_qiaozhe - 订单表
需要创建的表
* flight_info_qiaozhe - 航班信息表 (SQL脚本已准备)
API接口设计
认证接口 (/api/auth)
* POST /login - 用户登录
* POST /register - 用户注册
* POST /validate - 验证令牌
* POST /refresh - 刷新令牌
* POST /logout - 用户登出
* GET /me - 获取当前用户信息
订单接口 (/api/orders)
* POST / - 创建订单
* GET /{id} - 获取订单详情
* GET /my - 获取我的订单
* PUT /{id}/status - 更新订单状态 (管理员)
* PUT /{id}/cancel - 取消订单
* PUT /{id}/pay - 支付订单
* GET /search - 搜索订单 (管理员)
航班接口 (/api/flights)
* GET / - 获取所有航班
* GET /{id} - 获取航班详情
* GET /search - 综合搜索航班
* GET /bookable - 获取可预订航班
* POST / - 创建航班 (管理员)
* PUT /{id} - 更新航班 (管理员)
* DELETE /{id} - 删除航班 (管理员)

Commit:
feat: Implement JWT utility class for authentication and authorization

- Added JwtUtil class for generating, validating, and parsing JWT tokens.
- Integrated JWT secret and expiration time configuration in application.properties.
- Created SQL scripts for initializing the database with necessary tables and test data.
- Updated init.sql to create app_users_qiaozhe, flight_info_qiaozhe, and orders_qiaozhe tables with foreign key relationships.
- Enhanced application.properties with additional configurations for database, logging, and internationalization.

Commit:
feat: Implement JWT utility class for token generation and validation

- Added JwtUtil class for handling JWT operations including token creation, validation, and claims extraction.
- Integrated JWT secret and expiration time configuration in application.properties.
- Created SQL scripts for initializing the database with necessary tables and test data for users, flights, and orders.
- Updated init.sql to include comprehensive database setup and data insertion for the airline order management system.
- Enhanced application.properties with additional configurations for database connection, logging, and internationalization.

Commit:
Merge branch '20250728' into airline-order-course_day1-start

buoooou commented on backend/src/main/java/com/postion/airlineorderbackend/controller/OrderController.java:
```diff
@@ -0,0 +1,435 @@
+            return ResponseEntity.status(HttpStatus.CREATED).body(result);
+            
+        } catch (IllegalArgumentException e) {
+            log.warn("订单创建失败: {}", e.getMessage());
```
异常情况可以考虑使用全局异常处理来进行统一处理，返回的实体也封装一下

buoooou commented on backend/src/main/java/com/postion/airlineorderbackend/repository/FlightInfoRepository.java:
```diff
@@ -0,0 +1,229 @@
+ * @since 2024-01-01
+ */
+@Repository
+public interface FlightInfoRepository extends JpaRepository<FlightInfo, Long> {
```
非常好，对我们目前的讲述的系统功能上做了很丰富的优化和处理

buoooou commented on backend/src/main/java/com/postion/airlineorderbackend/service/AuthService.java:
```diff
@@ -0,0 +1,339 @@
+@RequiredArgsConstructor
+@Slf4j
+@Transactional
+public class AuthService {
```
虽然从实现来讲直接使用service也可以实现，但是在java或者说在springboot中，面向接口编程上来看，或者从springboot本身的ioc设计来看，直接不写接口只写实现并不是一个比较好的习惯，后续可以考虑先写接口后写实现，目前的业务逻辑比较简单，像我们今天的课程讲的内容，我们使用shedlock这些框架，有些分布式锁我们可以使用db，使用redis，框架内部怎么很方便的实现这个功能呢，也是先基于接口编程，后续实现可以用mysql实现，可以用redis实现，这些都是具体实现的内容，然后提供给外部框架的使用者，让他们可以用注解等方式来选择用db还是redis

Commit:
feat: add responsive navbar with authentication features

- Implemented a new navbar component with Angular Material.
- Added navigation links for orders, flights, and scheduled tasks based on user roles (admin/user).
- Included authentication buttons for login and registration.
- Integrated user dropdown menu displaying user information and logout option.
- Styled the navbar with SCSS for a modern look and responsive design.
- Created environment configuration for API URL.
- Set up main entry points for server-side rendering and client-side bootstrapping.
- Added global styles and TypeScript configuration files for the application.
- Included various image assets for the application.
